src/services/entity_builder.js

```diff
@@ -138,61 +136,40 @@ export default class EntityBuilder {
   validateAndCastFindEventsParams(params) {
     const allowedParametersList = ['assetId', 'fromTimestamp', 'toTimestamp', 'page', 'perPage', 'createdBy', 'data'];
 
-    validateFieldsConstrainedToSet(params, allowedParametersList);
+    const castedParams = validateAndCast(params)
+      .fieldsConstrainedToSet(allowedParametersList)
+      .castNumber(['fromTimestamp', 'toTimestamp', 'page', 'perPage'])
+      .isNonNegativeInteger(['fromTimestamp', 'toTimestamp', 'page', 'perPage'])
+      .isAddress(['createdBy'])
+      .getCastedParams();
 
     this.ensureGeoLocationParamsCorrectlyPlaced(params);
```

Comment:
Why isn't this also in `ValidationAndCasting `?

Reply:
ValidationAndCasting is designed for simple checks of individual fields, checking geolocation is a bit more complex